### PR TITLE
Add github-enterprise support

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #none# - documentation fixes and/or test additions
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG
* **Added** initial support for GitHub Enterprise (not integration tested yet)

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ const utils = require('./utils')
 // VCS implementations
 const BitbucketServer = require('./vcs/bitbucket-server')
 const GitHub = require('./vcs/github')
+const GitHubEnterprise = require('./vcs/github-enterprise')
```
Comment (sandersky) [inline lib/cli.js]: Does it make sense to require all of these or to do something like this below:
```js
try {
  return require(`./vcs/${provider}`)(config)
} catch (err) {
  throw new Error(`Unknown VCS provider: ${provider}`)
}
```

Comment (job13er) [inline lib/cli.js]: I guess with `require` that's fine, but I prefer to treat `require` like `import` and put them all at the top. In this case, I think those modules are small enough that there's not a big need to lazy-load it. If we end up with lots more providers, then yeah, maybe we want something like what you suggested.

**Comment by sandersky** [comment]
:+1:

**Comment by sglanzer** [comment]
👍 exciting!

**Pull request merged**